test/e2e/discovery/statefulsets.go

```diff
@@ -0,0 +1,269 @@
+			}
+		})
+	})
+
+	When("a pod tries to resolve a podname from stateful set in a local cluster", func() {
+		It("should resolve the pod IP from the local cluster", func() {
+			if !framework.TestContext.GlobalnetEnabled {
+				RunSSDiscoveryLocalTest(f)
+			} else {
+				framework.Skipf("Globalnet is enabled, skipping the test...")
+			}
+		})
+	})
+
+	When("the number of active pods backing a stateful set changes", func() {
+		It("should only resolve the IPs from the active pods", func() {
+			if !framework.TestContext.GlobalnetEnabled {
+				RunSSPodsAvailabilityTest(f)
+			} else {
+				framework.Skipf("Globalnet is enabled, skipping the test...")
+			}
+		})
+	})
+})
+
+func RunSSDiscoveryTest(f *lhframework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
+
+	By(fmt.Sprintf("Creating an Nginx Stateful Set on on %q", clusterBName))
+
+	nginxSSClusterB := f.NewNginxStatefulSet(framework.ClusterB)
+	appName := nginxSSClusterB.Spec.Selector.MatchLabels["app"]
+
+	By(fmt.Sprintf("Creating a Nginx Headless Service on %q", clusterBName))
+
+	nginxServiceClusterB := f.NewNginxHeadlessServiceWithParams(nginxSSClusterB.Spec.ServiceName, appName, framework.ClusterB)
+
+	f.NewServiceExport(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace)
+	f.AwaitServiceExportedStatusCondition(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace)
+
+	By(fmt.Sprintf("Creating a Netshoot Deployment on %q", clusterAName))
+
+	netshootPodList := f.NewNetShootDeployment(framework.ClusterA)
+
+	endpointSlices := f.AwaitEndpointSlices(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace, 1, 1)
+
+	verifyCount := 0
+
+	for _, endpointSlice := range endpointSlices.Items {
+		sourceCluster := endpointSlice.Labels[lhconstants.LabelSourceCluster]
+
+		for _, endpoint := range endpointSlice.Endpoints {
+			verifyEndpointsWithDig(f.Framework, framework.ClusterA, netshootPodList, endpoint, sourceCluster,
+				nginxServiceClusterB.Name, checkedDomains, true)
+			verifyCount++
+		}
+	}
+
+	Expect(verifyCount).To(Equal(1))
+	f.DeleteServiceExport(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace)
+	f.AwaitServiceImportCount(framework.ClusterA, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace, 0)
+
+	for _, endpointSlice := range endpointSlices.Items {
+		sourceCluster := endpointSlice.Labels[lhconstants.LabelSourceCluster]
+
+		for _, endpoint := range endpointSlice.Endpoints {
+			verifyEndpointsWithDig(f.Framework, framework.ClusterA, netshootPodList, endpoint, sourceCluster,
+				nginxServiceClusterB.Name, checkedDomains, false)
+		}
+	}
+}
+
+func RunSSDiscoveryLocalTest(f *lhframework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
```

Comment:
Comment completely unrelated to your patch.

I'm thinking that it'd be cleaner to encapsulate the context within the framework "f" itself. So we could do

f.Context.ClusterIDs[....]